File: src/main/java/com/yogesh/meditrack/service/MedicalRecordService.java
package com.yogesh.meditrack.service;

import com.yogesh.meditrack.entity.MedicalRecord;
import java.util.List;

public interface MedicalRecordService {

    MedicalRecord saveRecord(MedicalRecord record);

    List<MedicalRecord> getAllRecords();

    MedicalRecord getRecordById(Long id);

    MedicalRecord updateRecord(Long id,
                               MedicalRecord record);

    void deleteRecord(Long id);

}

File: frontend/src/service/medicalRecordService.js
import api from "./api";

export const getAllRecords = () => api.get("/medical-records");

export const getRecordById = (id) => api.get(`/medical-records/${id}`);

export const saveRecord = (record) => api.post("/medical-records", record);

export const updateRecord = (id, record) =>
  api.put(`/medical-records/${id}`, record);

export const deleteRecord = (id) => api.delete(`/medical-records/${id}`);